### PR TITLE
Add Launcher build stages to the release pipeline

### PR DESCRIPTION
**Pull request by christophermaier**
The pipeline will now ask if we need to build new Launcher artifacts
for this release. If we do, build steps will automatically be added;
if we don't, we'll just add the currently stable Launcher artifacts to
the release channel.

This adds automation that previously did not exist, and also *removes*
the manual step where we had to supply fully-qualified identifiers for
all Launcher packages we wanted to promote into the release
channel. Wins all around!

Signed-off-by: Christopher Maier <cmaier@chef.io>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by christophermaier** [comment]
Added the `DO-NOT-MERGE` label because there's still some testing that needs to be done, as well as addition of the Windows Service build.

File: .buildkite/launcher_build_steps.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      - docker#v2.0.0:
+          image: "chefes/windows-base:latest"
+          shell: [ "powershell", "-Command" ]
+          environment:
+            - HAB_BLDR_CHANNEL
+            - HAB_AUTH_TOKEN
+            - BUILDKITE_JOB_ID
+            - BUILDKITE_AGENT_ACCESS_TOKEN
+
+  - wait
```
Comment (mwrock) [inline .buildkite/launcher_build_steps.yaml]: Assuming each build is serialized? We should build the windows-service after the windows launcher.

File: .buildkite/launcher_build_steps.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    env:
+      HAB_BLDR_CHANNEL: unstable
+    plugins:
+      - docker#v2.0.0:
+          image: "chefes/windows-base:latest"
+          shell: [ "powershell", "-Command" ]
+          environment:
+            - HAB_BLDR_CHANNEL
+            - HAB_AUTH_TOKEN
+            - BUILDKITE_JOB_ID
```
Comment (mwrock) [inline .buildkite/launcher_build_steps.yaml]: please add `-HAB_CRYPTO_KEY`

File: .buildkite/launcher_build_steps.yaml (hunk 1 above)
Comment (christophermaier) [reply .buildkite/launcher_build_steps.yaml]: Yup, there's a space immediately after that line where we'll add it :) 

Need to coordinate with @smacfarlane for that, so it'll probably come after this PR.

**Pull request merged**